odoo/tools/mail.py

```diff
@@ -750,6 +751,20 @@ def formataddr(pair, charset='utf-8'):
     return f"{local}@{domain}"
 
 
+def formataddr_sanitized_name(pair, charset='utf-8'):
```

Comment:
@Julien00859 That famous " '@' in name issue " ... just tried a 'custom' formataddr tool sanitizing the name, aka: add spaces around '@' in name so that it is not considered as an email when trying to find emails in extract_rfc_2822. Went to a custom one because stable, and avoid breaking everything just for a hack ... any thoughts ?

Reply:
@Julien00859 zboing :kangaroo:

Reply:
@Julien00859 zboing :koala: